### PR TITLE
[StructuralMechanicsApplication] Finite strain kinematic plasti…

### PR DESCRIPTION
**Pull request by loumalouomega**
This PR adapts to the last changes the existing kinematic plasticity and adds the finite strain version of the kinematic plasticity

### CODE REVIEW

File: applications/StructuralMechanicsApplication/tests/cpp_tests/test_kinematic_plasticity_constitutive_laws_3d.cpp [hunk 1]
```diff
@@ -105,14 +108,98 @@ KRATOS_TEST_CASE_IN_SUITE(ConstitutiveLawIntegrateStressPlasticitySmallStrainKin
...
     TestVM = cl_parameters.GetStressVector();
     VonMisesCL.GetValue(PLASTIC_DISSIPATION, plastic_dissipation);
     KRATOS_WARNING_IF("TestPlasticity", plastic_dissipation < 1.0e-12) << "VonMises:: This test is not in plastic range" << std::endl;
-	
+
+
+    // Check the results
+    const double tolerance = 1.0e-4;
+    for (std::size_t comp = 0; comp < 6; ++comp){
+        KRATOS_CHECK_LESS_EQUAL(std::abs((VMres[comp] - TestVM[comp]) / VMres[comp]), tolerance);
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/tests/cpp_tests/test_kinematic_plasticity_constitutive_laws_3d.cpp:123]: there are now macros for comparing vectors

File: applications/StructuralMechanicsApplication/structural_mechanics_application.h [hunk 2]
```diff
@@ -587,6 +590,56 @@ class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) KratosStructuralMechanicsAppl
...
+    const GenericFiniteStrainKinematicPlasticity <HyperElasticIsotropicNeoHookean3D, GenericFiniteStrainConstitutiveLawIntegratorKinematicPlasticity<MohrCoulombYieldSurface<DruckerPragerPlasticPotential<6>>>> mHyperElasticNeoHookeanKinematicPlasticity3DMohrCoulombDruckerPrager;
+    const GenericFiniteStrainKinematicPlasticity <HyperElasticIsotropicNeoHookean3D, GenericFiniteStrainConstitutiveLawIntegratorKinematicPlasticity<MohrCoulombYieldSurface<TrescaPlasticPotential<6>>>> mHyperElasticNeoHookeanKinematicPlasticity3DMohrCoulombTresca;
+    const GenericFiniteStrainKinematicPlasticity <HyperElasticIsotropicNeoHookean3D, GenericFiniteStrainConstitutiveLawIntegratorKinematicPlasticity<TrescaYieldSurface<MohrCoulombPlasticPotential<6>>>> mHyperElasticNeoHookeanKinematicPlasticity3DTrescaMohrCoulomb;
+    const GenericFiniteStrainKinematicPlasticity <HyperElasticIsotropicNeoHookean3D, GenericFiniteStrainConstitutiveLawIntegratorKinematicPlasticity<DruckerPragerYieldSurface<MohrCoulombPlasticPotential<6>>>> mHyperElasticNeoHookeanKinematicPlasticity3DDruckerPragerMohrCoulomb;
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/structural_mechanics_application.h:641]: @KratosMultiphysics/technical-committee this should FINALLY be solved better, this makes developing in StructuralMechanics more and more difficult

**Review by AlejandroCornejo — APPROVED** [review]
Yes we should think how to reduce the number of instantiations...

**Comment by loumalouomega** [comment]
> Yes we should think how to reduce the number of instantiations...

I was changing the vector stuff but Ok, another PR

**Pull request merged**